src/thor/bidirectional_astar.cc

```diff
@@ -895,15 +897,22 @@ void BidirectionalAStar::SetDestination(GraphReader& graphreader,
     }
     // Get the directed edge
     graph_tile_ptr tile = graphreader.GetGraphTile(edgeid);
+    if (tile == nullptr) {
+      continue;
+    }
     const DirectedEdge* directededge = tile->directededge(edgeid);
 
     // Get the opposing directed edge, continue if we cannot get it
-    GraphId opp_edge_id = graphreader.GetOpposingEdgeId(edgeid);
```

Comment:
instead of the other changes below i would simplify this to this right at the beginning:

```c++
// Get the opposing directed edge, continue if we cannot get it
graph_tile_ptr opp_tile = nullptr;
const DirectedEdge* opp_dir_edge = nullptr;
auto opp_edge_id = graphreader.GetOpposingEdgeId(edgeid, opp_dir_edge, opp_tile);
if (!opp_dir_edge)  {
  continue;
}
```

then you dont need any more other checks and continues, just use the variables if it gets passed this first check